src/Core/Intervals.php

```diff
@@ -0,0 +1,203 @@
+
+        // should return id be quoted or not
+        if ($quotes) {
+            $seedId = "'" . $seedId . "'";
+        }
+
+        return $seedId;
+    }
+
+    /**
+     * Generate a 'parent' id for use
+     * by handleType:'parent'
+     *
+     * @param int $counter
+     * @param string $curModule
+     * @param string $relModule
+     *
+     * @return string
+     */
+    public function getRelatedUpId($counter, $curModule, $relModule)
+    {
+        /* The relModule that we point up to should be the base */
+        return $this->getRelatedId($counter, $curModule, $relModule, $relModule);
+    }
+
+    /**
+     * Generate a 'related' id for use
+     * by handleType:'related' and 'generateRelationships'
+     *
+     * @param int $counter
+     * @param string $curModule
+     * @param string $relModule
+     * @param string $baseModule
+     *
+     * @return integer
+     */
+    public function getRelatedId($counter, $curModule, $relModule, $baseModule)
+    {
+        if (empty($this->counters[$curModule . $relModule])) {
+            $this->counters[$curModule . $relModule] = 0;
+        }
+
+        $c = $this->counters[$curModule . $relModule];
+        $result = 0;
+
+        $modules = $this->config->get('modules');
+
+        /*
+         * All ratios can be determined simply by looking
+         * at the relative counts in $GLOBALS['modules']
+         *
+         * Such a module must exist.
+         */
+        if (!empty($modules[$relModule])) {
+            $baseToRelatedRatio = $modules[$relModule] / $modules[$baseModule];
+            $baseToThisRatio = $modules[$curModule] / $modules[$baseModule];
+
+            /*
+             * floor($this->count/$acctToThisRatio) determines what 'group'
+             * this record is a part of.  Multiplying by $acctToRelatedRatio
+             * gives us the starting record for the group of the related type.
+             */
+            $n = floor(floor($counter / $baseToThisRatio) * $baseToRelatedRatio);
+
+            $this->counters[$curModule . $relModule]++;
+
+            /*
+             * There are $acctToRelatedRatio of the related types
+             * in the group, so we can just pick one of them.
+             */
+            $result = $n + ($c % ceil($baseToRelatedRatio));
+        }
+
+        return $result;
+    }
+
+    /**
+     * Returns an alias to be used for id generation. Always honor the
+     * configured alias if one exists, otherwise for longer module names (over
+     * 10 chars), use the first and last 5 characters of the passed-in name
+     * (even if they overlap).
+     *
+     * @param string $moduleName
+     * @return string
+     */
+    public function getAlias($moduleName)
+    {
+        $aliases = $this->config->get('aliases');
+
+        if (isset($aliases[$moduleName])) {
+            return $aliases[$moduleName];
+        } elseif (strlen($moduleName) > 10) {
+            return substr($moduleName, 0, 5) . substr($moduleName, -5);
```

Comment:
mmm ... may be constant? 5? =)
Besides, that if in this case we will get equal modules aliases?